dashboard/test/controllers/api/v1/pd/application/teacher_applications_controller_test.rb

```diff
@@ -149,13 +169,48 @@ class TeacherApplicationsControllerTest < ::ActionController::TestCase
       Pd::Application::TeacherApplicationMailer.expects(:principal_approval).never
 
       sign_in @applicant
-      put :create, params: {form_data_hash: @test_params, status: 'incomplete'}
+      put :create, params: {form_data_hash: @test_params, isSaving: true}
       refute TEACHER_APPLICATION_CLASS.last.response_scores
       assert_response :created
     end
 
-    test 'autoscores and queues email once application is submitted' do
-      application_hash = build :pd_teacher_application_hash_common, :csp
+    test 'autoscores and queues emails on submit when approval is required' do
```

Comment:
There's a lot of repeated code in these new tests. I wrote up a `.each` way to reduce duplication, but I found the readability hard. Please push back if you think this is too much!

Reply:
I feel like prioritizing readability might make sense since we are shifting lots of terminology around statuses and desired behaviors so having each one more explicitly laid out might be better as we home in on the desired outcome? That's just a thought though, I'm not dug in on this by any means!

Reply:
I'm on the same page –– thanks!